File: spec/functional_test/fixtures/perl/cli_getopt/tool.pl
use strict;
use warnings;
use Getopt::Long;

my $port;
my $verbose;
GetOptions(
    "port=i"  => \$port,
    "verbose" => \$verbose,
);

my $token = $ENV{API_TOKEN};
my $first = $ARGV[0];
print "$port $verbose $token $first\n";

File: spec/functional_test/fixtures/scala/cli_scallop/tool.scala
import org.rogach.scallop._

class Conf(arguments: Seq[String]) extends ScallopConf(arguments) {
  val verbose = opt[Boolean]("verbose")

  val serve = new Subcommand("serve") {
    val port = opt[Int]("port")
    val file = trailArg[String]()
  }
  addSubcommand(serve)
  verify()
}

object Tool {
  def main(args: Array[String]): Unit = {
    val conf = new Conf(args)
    val token = sys.env("API_TOKEN")
    println(token)
  }
}
